frontend/src/types/errorResponse.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * An error response returned by the API. This implements the error responses
+ * ADR, please refer to doc/adr/0012-error-responses.md for more information.
+ */
+export type ErrorResponse<
+  T extends keyof ErrorResponseDetail = "__fallback__",
+  Children = never,
+> = {
+  /**
+   * Type of the error.
+   */
+  type: T
+
+  /**
+   * Title of the error. This is intended for debugging purposes and should
+   * not be parsed or displayed in the UI.
+   */
+  title?: string
+
+  /**
+   * Details of the error. This is intended for debugging purposes and should
+   * not be parsed or displayed in the UI.
+   */
+  details?: string
+
+  /**
+   * Instance of the error. This can be used for matching errors to specific
+   * locations in the UI (e.g. an invalid form field).
+   */
+  instance?: string
+
+  /**
+   * Nested errors. If an error response contains multiple messages (e.g.
+   * for listing all fields that failed validation of a form), they will
+   * appear here.
+   */
+  errors?: Array<Children>
```

Comment:
Shouldn't the childen errors be of type themselves `ErrorDetail`? Not sure if it is correct but I remember that we say that the errors themselves are of the same type as the upper parent?